pkg/compute/build.go

```diff
@@ -23,6 +23,12 @@ const IgnoreFilePath = ".fastlyignore"
 
 // Toolchain abstracts a Compute@Edge source language toolchain.
 type Toolchain interface {
+	Name() string
+	DisplayName() string
+	StarterKits() []StarterKit
+	SourceDirectory() string
+	IncludeFiles() []string
+	Initialize(out io.Writer) error
 	Verify(out io.Writer) error
 	Build(out io.Writer, verbose bool) error
 }
```

Comment:
I don't like the use of interface methods here to model getters for simple values. Would this be better modelled as a `Language` struct with an embedded `Toolchain` interface? Such as:

```suggestion
type Toolchain interface {
	Initialize(out io.Writer) error
	Verify(out io.Writer) error
	Build(out io.Writer, verbose bool) error
}

type Language struct {
    Name string
    DisplayName string
    StarterKits []StarterKit
    IncludeFiles []string
    
    Toolchain
}

func NewLangauge(name, displayName, string, kits []starterKits, includeFiles []string, toolchain Toolchain) *Language {
    // ... etc
}
```

Reply:
`NewLangauge` -> `NewLanguage` in your suggestion :)

I personally like the Language struct idea because this `NewLanguage` function could be where the "metadata"-y bits that I was talking about could be initialized and verified.

Reply:
@phamann I like your alternative — structs collect data, and interfaces collect behavior, so your type Language struct seems to best model what's actually going on.

Reply:
Ok @mccurdyc @peterbourgon Have implemented this in https://github.com/fastly/cli/pull/160/commits/5d59eb8165d13bb13d31b08e488f7080b0db7ae8 let me know your thoughts?